lib/gobl/operations.rb

```diff
@@ -0,0 +1,109 @@
+module GOBL
+  module Operations
+    #
+    # Calculates and validates a document or envelop, wrapping it in an envelope if needed
+    #
+    # @param struct [GOBL::Document, GOBL::Envelope] the document or an envelope with the document to
+    #   build
+    # @param envelop [Boolean] if the struct is a document, whether the operation will wrap it in an
+    #   envelope
+    # @param draft [Boolean] if the return value is a envelope, whether that envelope should be
+    #   flagged as a draft
+    #
+    # @return [GOBL::Envelope, GOBL::Document] a calculated and validated envelope or document
+    #   (depending on the other parameters)
+    #
+    # @raise [GOBL::InvalidDocumentException] if the document is invalid after the calculation.
+    #
+    # @example Build a document without enveloping it
+    #   invoice = GOBL::Bill::Invoice.new(…)
+    #   GOBL.build(invoice) #=> A new, calculated GOBL::Bill::Invoice
+    #
+    # @example Build a document and wrap in a draft envelope
+    #   invoice = GOBL::Bill::Invoice.new(…)
+    #   GOBL.build(invoice, envelop: true) #=> A calculated, draft GOBL::Envelope
+    #
+    # @example Build a document and wrap in a non-draft envelope
+    #   invoice = GOBL::Bill::Invoice.new(…)
+    #   GOBL.build(invoice, envelop: true, draft: false) #=> A calculated, non-draft GOBL::Envelope
+    #
+    # @example Build an envelope
+    #   envelope = GOBL::Envelop.new(…)
+    #   GOBL.build(envelope) #=> A new, calculated GOBL::Envelope
+    #
+    # @example Build an envelope forcing it to be a non-draft
+    #   envelope = GOBL::Envelop.new(…)
+    #   GOBL.build(envelope, draft: false) #=> A new, non-draft GOBL::Envelope
+    #
+    def build(struct, envelop: false, draft: true)
```

Comment:
This makes sense. We want to "deter" usage of `build(invoice, envelop: true, draft: false)` as ideally users should use the sign method.